src/components/pages/DataVisualizations/YearLimitsSelect.jsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Button, Input, getFieldValue } from 'antd';
+import {
+  setVisualizationData,
+  resetVisualizationQuery,
+  setHeatMapYears,
+} from '../../../state/actionCreators';
+import YearLimitsSlider from './YearLimitsSlider';
+import { rawApiDataToPlotlyReadyInfo, useInterval } from '../../../utils';
+import reducers from '../../../state/reducers';
+
+import Redux, { createStore } from 'redux';
+import { connect } from 'react-redux';
+
+const store = createStore(reducers);
+
+const mapStateToProps = (state, ownProps) => {
+  const { view, office } = ownProps;
+  if (office === 'all' || !office) {
+    switch (view) {
+      case 'time-series':
+        return {
+          years: state.vizReducer.timeSeriesAllYears,
+        };
+      case 'office-heat-map':
+        return {
+          years: state.vizReducer.officeHeatMapYears,
+        };
+      case 'citizenship':
+        return {
+          years: state.vizReducer.citizenshipMapAllYears,
+        };
+      default:
+        return {
+          years: ['', ''],
+        };
+    }
+  } else {
+    switch (view) {
+      case 'time-series':
+        return {
+          years: state.vizReducer.offices[office].timeSeriesYears,
+        };
+      case 'citizenship':
+        return {
+          years: state.vizReducer.offices[office].citizenshipMapYears,
+        };
+      default:
+        return {
+          years: ['', ''],
+        };
+    }
+  }
+};
+
+function YearLimitsSelect(props) {
+  let { view, office, dispatch, clearQuery, updateStateWithNewData, years } = props;
+  const yearInputsOnChange = (view, office, e) =>
+    dispatch(
+      setHeatMapYears(
+        view,
+        office,
+        e.target.id.includes('year_start') ? 0 : 1,
+        e.target.value
+      )
+    );
+  const stateSettingFn = (view, office, data) => {
+    const plotlyReadyData = rawApiDataToPlotlyReadyInfo(view, office, data);
+    dispatch(setVisualizationData(view, office, plotlyReadyData));
+  };
+  const [form] = Form.useForm();
+  useInterval(() => {
+    form.setFieldsValue({
+      year_start: years[0],
+      year_end: years[1],
+    });
+  },1000);
+  return (
+    <div
+      className="year-limits-select-container"
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        minHeight: '50px',
+      }}
+    >
+      <YearLimitsSlider
+        office={office}
+        view={view}
+        lowerLimit={2015}
+        upperLimit={2022}
+        step={1}
+      />
+      <Form
+        form={form}
+        name="yearLimitsSelect"
+        initialValues={{ year_start: years[0], year_end: years[1] }}
+        onFinish={() => {
```

Comment:
LOVE this design pattern. Functional programming is a great way to organize and think about the work that these components need in order to display the data that the user wants to see. 

There are a few ways we can optimize some of this but for now this shows an extreme level of engineering effort and ability.

Reply:
Thank you!

One question: Ash had mentioned checking out a branch with the "feature/BL-790-fe" prefix, and replacing this draft PR with one from the properly prefixed branch. Should I go ahead with that or leave this [ branch name / PR ] for consistency?

Reply:
For the Jira ticket connection, it's important to have the Jira issue key at the beginning of the PR. I would change the branch name over to the properly prefixed as you were saying.